mesos_collectd.py

```diff
@@ -142,77 +176,127 @@ def configure_callback(conf, is_master, prefix, cluster, instance, path, host,
     CONFIGS.append({
         'host': host,
         'port': port,
-        'mesos_url': "http://" + host + ":" + str(port) + "/metrics/snapshot",
-        'framework_url': ("http://" + host + ":" + str(port) +
+        'mesos_url': http_prefix + host + ":" + str(port) + "/metrics/snapshot",
+        'framework_url': (http_prefix + host + ":" + str(port) +
                           "/master/frameworks"),
-        'task_url': "http://" + host + ":" + str(port) + "/master/tasks",
+        'task_url': http_prefix + host + ":" + str(port) + "/master/tasks",
         'system_health_url': system_health_url,
         'verboseLogging': verboseLogging,
         'version': version,
         'instance': instance,
         'cluster': cluster,
         'path': path,
+        'http_prefix': http_prefix,
+        'dcos_sfx_username': dcos_sfx_username,
+        'dcos_sfx_password': dcos_sfx_password,
+        'dcos_auth_token': dcos_auth_token,
+        'master_url': master_url,
+        'dcos_auth_header': dcos_auth_header,
+        'ca_file_path': ca_file_path,
+        'ssl_context': ssl_context
     })
 
-
-def get_version_from_api(host, port):
-    version_api_url = "http://" + host + ":" + str(port) + "/version"
+def get_dcos_auth_token(uid, password, host, master_url):
     try:
-        result = json.load(urllib2.urlopen(version_api_url, timeout=10))
+        collectd.info('INFO: Getting DC/OS authentication token.')
+        headers = {"Content-Type":"application/json"}
+        data = ('{"uid":"%s","password":"%s"}' % (uid, password))
```

Comment:
Use `json.dumps` because the current code will break if password has a double quote in it.